src/editor/Gui.cpp

```diff
@@ -586,22 +623,81 @@ void Gui::drawMenuBar() {
 				if (map)
 				{
 					logf("Export wad: %s%s%s\n", g_settings.gamedir.c_str(), g_settings.workingdir.c_str(), (map->name + ".wad").c_str());
-					ExportWad(map);
-					logf("Remove all embedded textures\n");
-					map->delete_embedded_textures();
-					if (map->ents.size())
+					if (ExportWad(map))
 					{
-						std::string wadstr = map->ents[0]->keyvalues["wad"];
-						if (wadstr.find(map->name + ".wad" + ";") == std::string::npos)
+						logf("Remove all embedded textures\n");
+						map->delete_embedded_textures();
+						if (map->ents.size())
+						{
+							std::string wadstr = map->ents[0]->keyvalues["wad"];
+							if (wadstr.find(map->name + ".wad" + ";") == std::string::npos)
+							{
+								map->ents[0]->keyvalues["wad"] += map->name + ".wad" + ";";
+							}
+						}
+					}
+				}
+			}
+
+			if (ImGui::MenuItem("Wavefront (.obj)", NULL)) {
+				Bsp* map = app->getMapContainingCamera()->map;
+				if (map)
+				{
+					map->ExportToObjWIP(g_settings.gamedir + g_settings.workingdir);
+				}
+			}

```

Comment:
Rather than adding WIP to the method names, can you add WIP to the menu item label? The point is to make it clear to users that the feature isn't finished and might not work as expected.

Also would be nice to show a tooltip explaining what is/isn't working with this feature. Is it just textures and UV coordinates that are missing?

Reply:
Yes need add export materials and write coordinates to obj file, i can try do it later. Now it just export map geometry